extensions/blocks/contact-form/index.js

```diff
@@ -69,7 +69,9 @@ export const settings = {
 	},
 
 	edit: JetpackContactForm,
-	save: InnerBlocks.Content,
```

Comment:
It does seem to me that this is a valid usage as we always considered `save` as a "component". Should we update the check function @gziolo?

Reply:
In `getSaveElement ` we have https://github.com/WordPress/gutenberg/blob/master/packages/blocks/src/api/serializer.js#L68-L70:

```js
	// Component classes are unsupported for save since serialization must
	// occur synchronously. For improved interoperability with higher-order
	// components which often return component class, emulate basic support.
```

We provide a workaround, so we should relax this check. I will open PR soon.

Reply:
It seems like this can't be the issue as we have the same check for `save` method almost forever:
https://github.com/WordPress/gutenberg/blob/da8184a055d8b2d40ba0306faf7103c0a961726a/blocks/api/registration.js#L96-L101
```js
if ( ! settings || ! isFunction( settings.save ) ) {
```

Reply:
@gziolo Oh, that's a bigger issue that we may even consider for `5.2.x`

Reply:
Opened https://github.com/WordPress/gutenberg/pull/16152 to address the issue where statics were not hoisted for deprecated components. This should fix the issue which happens here.

Reply:
In my testing, you will have to use function expression for this block for safety. It's fragile to assume that `InnerBlocks.Content` will always be a function. The same applies to all `save` implementations. Even if we allow objects, it still might work differently than intended.

Related Slack discussion on WordPress.org: https://wordpress.slack.com/archives/C02QB2JS7/p1560500199068200

Reply:
@gwwar I think we might still need this change in JetPack because the component is not a function anymore.